src/mainloop.c

```diff
@@ -74,11 +76,26 @@ static int sock_set_non_blocking(int fd)
 
 static int get_port_number(char *arg_port)
 {
-    int port_number = strtol(arg_port, &arg_port, 10);
-    if (port_number <= 0 || port_number > 65535) {
-        port_number = DEFAULT_PORT_NUMBER;
+    long ret;
+    char *endptr;
+
+    ret = strtol(arg_port, &endptr, 10);
+    if ((errno == ERANGE && (ret == LONG_MAX || ret == LONG_MIN)) ||
+        (errno != 0 && ret == 0)) {
+        perror("strtol");
```

Comment:
Show verbose messages.

Reply:
It is not meaningful to show "strtol" here. Instead, you should dump something like "Failed to parse port number"

Reply:
Don't call `perror` here. Verbose message is expected to appear before termination.

Reply:
I don't know how c language thinks about error handling. Suppose the port number entered by the user today is out of range, should we just exit the program immediately with error message or set the port number to default and print the error message but continue to run?